chore: enhance migration guide (#1435)
containers/podman-desktop-extension-ai-lab: MIGRATION.md

Version `1.0` of the catalog adds an important property to models `backend`, defining the type of framework required by the model to run (E.g. LLamaCPP, WhisperCPP).

### 🛠️ How to migrate

You can either delete any existing `user-catalog` by deleting the `~/.local/share/containers/podman-desktop/extensions-storage/redhat.ai-lab/user-catalog.json`.

> [!WARNING]  
> This will remove the models you have imported from the catalog. You will be able to import it again afterward.

If you want to keep the data, you can migrate it by updating certain properties.

### Recipes

The recipe object has a new property `backend` which define which framework is required. 
Value accepted can be `llama-cpp`, `whisper-cpp`, `none`.

> Moreover, the `models` property has been changed to `recommended`.

Reviewer: We should explain the change of semantic between models and recommended and what is happening when recommended is absent

Author: :+1:

**Example**

```diff
{
  "version": "1.0",
  "recipes": [{
    "id": "chatbot",
    "description" : "This is a Streamlit chat demo application.",
    "name" : "ChatBot",
    "repository": "https://github.com/containers/ai-lab-recipes",
-   "models": [
+   "recommended": [
      "hf.instructlab.granite-7b-lab-GGUF",
       "hf.instructlab.merlinite-7b-lab-GGUF"
    ]
+   "backend": "llama-cpp"
  }],
  "models": [],
  "categories": []
}
```

### Models

The model object has also the new property `backend`, which define which framework is required.

> We increase the security by adding a new optional `sha256` property.

Reviewer: Would be good to have a code snippet explaining how to compute the value from the file using sha256sum

Author: :+1:

Version `1.0` of the catalog adds an important property to models `backend`, defining the type of framework required by the model to run (E.g. LLamaCPP, WhisperCPP).

### 🛠️ How to migrate

> You can either delete any existing `user-catalog` by deleting the `~/.local/share/containers/podman-desktop/extensions-storage/redhat.ai-lab/user-catalog.json`.

Reviewer 2: ```suggestion
You can optionally delete any existing `user-catalog` by removing the file `~/.local/share/containers/podman-desktop/extensions-storage/redhat.ai-lab/user-catalog.json`.
```

> [!WARNING]  
> This will remove the models you have imported from the catalog. You will be able to import it again afterward.

> If you want to keep the data, you can migrate it by updating certain properties.

Reviewer 2: ```suggestion
If you want to keep the data, you can migrate it by updating certain properties within the recipes and models fields.
```

### Recipes

> The recipe object has a new property `backend` which define which framework is required. 

Reviewer 2: ```suggestion
The recipe object has a new property `backend` which defines which framework is required. 
```

> Value accepted can be `llama-cpp`, `whisper-cpp`, `none`.

Reviewer 2: ```suggestion
Value accepted are `llama-cpp`, `whisper-cpp` and `none`.
```

Moreover, the `models` property has been changed to `recommended`.

> [!TIP]
> Before Podman AI Lab version v1.2 recipes uses the `models` property to list the models compatible. Now all models using the same `backend` could be used. We introduced `recommended` to highlight certain models. 

**Example**

```diff
{
  "version": "1.0",
  "recipes": [{
    "id": "chatbot",
    "description" : "This is a Streamlit chat demo application.",
    "name" : "ChatBot",
    "repository": "https://github.com/containers/ai-lab-recipes",
-   "models": [
+   "recommended": [
      "hf.instructlab.granite-7b-lab-GGUF",
       "hf.instructlab.merlinite-7b-lab-GGUF"
    ]
+   "backend": "llama-cpp"
  }],
  "models": [],
  "categories": []
}
```

### Models

> The model object has also the new property `backend`, which define which framework is required.

Reviewer 2: ```suggestion
The model object has also the new property `backend`, which defines which framework is required.
```

> We increase the security by adding a new optional `sha256` property.

Reviewer 2: ```suggestion
Additionally, we have enhanced security by introducing a new optional `sha256` property.
```
